### PR TITLE
Add native binding

### PR DESCRIPTION
**Pull request by lmoe**
# Description of change

Adds a new binding that can be compiled into a common shared library. 
It can be referenced by other languages such as Go/C/C#/etc.

It exports the following functions:

```
init_logger
binding_get_last_error

call_client_method
call_secret_manager_method
call_utils_method
call_wallet_method

create_client
create_secret_manager
create_wallet
listen_wallet

destroy_client
destroy_secret_manager
destroy_string
destroy_wallet

get_client_from_wallet
get_secret_manager_from_wallet
```

C/C++ example headers were generated and are to be found at `bindings/native/headers`.

A Go library example can be found [here](https://github.com/iotaledger/wasp-wallet-sdk/tree/main)
The Go library is primarily used for the secret managers, but a basic Wallet implementation exists as well.

A readme was added in `bindings/native/README.md` to explain string/error handling, but it's not a complete guide.

A CI workflow was added to build the library for linux, mac and windows. A trigger workflow does not exist yet.

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Tests were written in Go and can be found [here](https://github.com/iotaledger/wasp-wallet-sdk/tree/main/test)

### CODE REVIEW

File: .github/workflows/bindings-native-publish.yml
Comment (thibault-martinez) [inline .github/workflows/bindings-native-publish.yml:1]: Does this actually publish anything?

Comment (lmoe) [reply .github/workflows/bindings-native-publish.yml:1]: I'm not super well versed with those workflows. 
I've tested it in a private repo and I made it build and create downloadable artifacts.
 
Where they should be published to or anything else, I don't know how you want that handled and figured you know it better than I do. :) 

If you give me some instructions I might be able to complete the workflow on my own.

**Comment by thibault-martinez** [comment]
Can you target the 1.1 branch please 🙏🏻

**Comment by wireless90** [comment]
This would be useful for the c# binding. Hope it gets merged soon thanks

**Comment by lmoe** [comment]
@wireless90 Happy to see C# being used! 

@thibault-martinez Rebased with 1.1, Go tests currently running locally. Will test it with wasp-cli as well and give you feedback afterwords.

**Comment by thibault-martinez** [comment]
Will be reopened in https://github.com/iotaledger/iota-sdk-native-bindings

**Pull request closed without merging**